src/app/api/services/edit-page/edit-page.service.ts

```diff
@@ -3,22 +3,151 @@ import { Injectable } from '@angular/core';
+import { PageMode } from '../../../portlets/dot-edit-content/components/dot-edit-page-toolbar/dot-edit-page-toolbar.component';
+import { DotEditPageState } from '../../../shared/models/dot-edit-page-state/dot-edit-page-state.model';
+import { DotRenderedPageState } from '../../../portlets/dot-edit-page/shared/models/dot-rendered-page-state.model';
 
 /**
  * Provide util methods to get a edit page html
  * @export
  * @class EditPageService
  */
+
 @Injectable()
 export class EditPageService {
     constructor(private coreWebService: CoreWebService) {}
 
-    get(url: string): Observable<DotRenderedPage> {
+    /**
+     * Get the page HTML in edit mode
+     *
+     * @param {string} url
+     * @returns {Observable<DotRenderedPage>}
+     * @memberof EditPageService
+     */
+    getEdit(url: string): Observable<DotRenderedPage> {
+        return this.get(url, 'EDIT_MODE');
+    }
+
+    /**
+     * Get the page HTML in preview mode
+     *
+     * @param {string} url
+     * @returns {Observable<DotRenderedPage>}
+     * @memberof EditPageService
+     */
+    getPreview(url: string): Observable<DotRenderedPage> {
+        return this.get(url, 'PREVIEW_MODE');
+    }
+
+    /**
+     * Get the page HTML in live mode
+     *
+     * @param {string} url
+     * @returns {Observable<DotRenderedPage>}
+     * @memberof EditPageService
+     */
+    getLive(url: string): Observable<DotRenderedPage> {
+        return this.get(url, 'LIVE_MODE');
+    }
+
+    /**
+     * Lock a content asset
+     *
+     * @param {string} inode
+     * @returns {Observable<any>}
+     * @memberof PageViewService
+     */
+    lock(inode: string): Observable<any> {
+        return this.coreWebService.requestView({
+            method: RequestMethod.Put,
+            url: `content/lock/inode/${inode}`
+        });
+    }
+
+    /**
+     * Unlock a content asset
+     *
+     * @param {string} inode
+     * @returns {Observable<any>}
+     * @memberof PageViewService
+     */
+    unlock(inode: string): Observable<any> {
+        return this.coreWebService.requestView({
+            method: RequestMethod.Put,
+            url: `content/unlock/inode/${inode}`
+        });
+    }
+
+    /**
+     * Set the page state
+     *
+     * @param {DotRenderedPage} page
+     * @param {DotEditPageState} state
+     * @returns {Observable<any>}
+     * @memberof EditPageService
+     */
+    setPageState(page: DotRenderedPage, state: DotEditPageState): Observable<DotRenderedPageState> {
+
+        const methods = [];
+        let lockUnlock;
+        let pageMode;
+
+        // When lock prop is not set is null
+        if (state.lock === true) {
+            lockUnlock = this.lock(page.liveInode)
+                .pluck('bodyJsonObject')
+                .pluck('message');
+        } else if (state.lock === false) {
+            lockUnlock = this.unlock(page.liveInode)
+                .pluck('bodyJsonObject')
+                .pluck('message');
+
+        }
```

Comment:
maybe we should create a private method for this

Reply:
Done

Reply:
lockUnlock = (state.lock === true)?this.lock(page.liveInode).pluck('bodyJsonObject').pluck('message'):
                                                          this.unlock(page.liveInode).pluck('bodyJsonObject').pluck('message');

Reply:
Not exactly, because if ```state.lock``` is undefined we don't want to do any.